### PR TITLE
Issue #48 by Inlead: Possibility of inserting h2 and h3.

### PR DESCRIPTION
**Pull request by guddo**
http://platform.dandigbib.org/issues/48

### CODE REVIEW

File: modules/ding_content/ding_content.features.wysiwyg.inc [hunk 1]
```diff
@@ -51,20 +54,18 @@ function ding_content_wysiwyg_default_profiles() {
...
+      'resize_enabled' => 1,
+      'default_toolbar_grouping' => 0,
+      'simple_source_formatting' => 0,
+      'acf_mode' => 0,
+      'acf_allowed_content' => '',
       'css_setting' => 'self',
       'css_path' => '/profiles/ding2/themes/ddbasic/css/ddbasic.wysiwyg.css',
-      'css_classes' => '',
+      'stylesSet' => '',
+      'block_formats' => 'p,address,pre,h2,h3,h4,h5,h6,div',
```
Comment (kasperg) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: Supporting all these formats seems excessive compared to the original issue. Besides making things more complex I also fear that all these headings have not been styled properly.

How about reducing this to `p,h2,h3,div`

File: modules/ding_content/ding_content.features.wysiwyg.inc [hunk 2]
```diff
@@ -51,20 +54,18 @@ function ding_content_wysiwyg_default_profiles() {
...
-      'remove_linebreaks' => 1,
-      'apply_source_formatting' => 0,
-      'paste_auto_cleanup_on_paste' => 0,
-      'block_formats' => 'p,address,pre,h2,h3,h4,h5,h6,div',
+      'toolbarLocation' => 'top',
+      'resize_enabled' => 1,
+      'default_toolbar_grouping' => 0,
+      'simple_source_formatting' => 0,
+      'acf_mode' => 0,
+      'acf_allowed_content' => '',
```
Comment (kasperg) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: There are a range of seemingly unrelated changes here. Some seem important e.g. `verify_html` and `remove_linebreaks`.

Is this intentional or perhaps a byproduct of updating the WYSIWYG module without updating the configuration accordingly?

**Comment by kasperg** [comment]
Reviewed. I have a couple of comments.

**Comment by kasperg** [comment]
Also: There are two other aspects of this mentioned in the issue:

- Styling of the headings (`h2` and `h3`)
- Handling of existing content

How does this change relate to those? I see no CSS changes as a part of this PR.

File: modules/ding_content/ding_content.features.wysiwyg.inc (hunk 2 above)
Comment (guddo) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: This is a byproduct of contrib module upgrade. Actually in our last Sprint we experienced a lot of changes in the default export that we made, and had to manually adjust it in order not to make "unrelated" changes in our commits.
With this said, I think that core team should make exports of everything after next release - in order to keep exports up to date with updated contrib modules, and in order to not have these mixed up commits from various contributors.

File: modules/ding_content/ding_content.features.wysiwyg.inc (hunk 1 above)
Comment (guddo) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: We have not changed any of the already defined formats. As you can see L64 in previous commit is identical.

**Comment by guddo** [comment]
Headlines are already defined in `themes/ddbasic/sass/framework/_base.scss`.

File: modules/ding_content/ding_content.features.wysiwyg.inc (hunk 1 above)
Comment (kasperg) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: As I understand it these formats were not available to the editor user through the WYSIWYG editor as the `Format` setting was not enabled?

File: modules/ding_content/ding_content.features.wysiwyg.inc (hunk 2 above)
Comment (kasperg) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: Noted. [I have created a ticket for this](http://platform.dandigbib.org/issues/1551).

**Comment by kasperg** [comment]
I can see that the sizes are defined in the stylesheet. I am unsure whether they are appropriatly styled/sized.

[One of the editors write](http://platform.dandigbib.org/issues/48#note-9): "Request: A h2-headline, which in the stylesheet has been formatted so it is smaller than the headline for the page [I assume this is h1]  and does not appear a line above the text where it belongs [?]".

How does this pull request relate to this?

File: modules/ding_content/ding_content.features.wysiwyg.inc (hunk 1 above)
Comment (guddo) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: That is correct.

Ok, so just allow `p,h2,h3,div`?

Comment (kasperg) [inline modules/ding_content/ding_content.features.wysiwyg.inc]: That makes sense to me based on the info in the ticket.

**Comment by guddo** [comment]
It has been verified that all font-sizes are properly defined in the theme variables.

If there is issues where the Drupal page/pane-title is smaller than the usual h1 (and the h2 thereby larger), it should be reported as a separate issue.

**Comment by kasperg** [comment]
Thanks for the followup @guddo. With that in mind. Reviewed, approved and awaiting merge.

**Comment by kasperg** [comment]
Merged. Thanks!
